src/cl/basic/cl_basic_context.c

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "cl_basic.h"
+#include "cl/ucc_cl_log.h"
+#include "utils/ucc_malloc.h"
+
+UCC_CLASS_INIT_FUNC(ucc_cl_basic_context_t,
+                    const ucc_base_context_params_t *params,
+                    const ucc_base_config_t *config)
+{
+    const ucc_cl_context_config_t *cl_config =
+        ucc_derived_of(config, ucc_cl_context_config_t);
+    UCC_CLASS_CALL_SUPER_INIT(ucc_cl_context_t, cl_config->cl_lib);
+    cl_info(cl_config->cl_lib, "initialized cl context: %p", self);
+    return UCC_OK;
+}
+
+UCC_CLASS_CLEANUP_FUNC(ucc_cl_basic_context_t)
```

Comment:
Minor: Should we use INIT and FINALIZE, instead of INIT and CLEANUP.  Probably, you are following UCX style.

Reply:
I would keep macros as is. We reuse UCS class functionality. We could rename. But for "finalize" for class destructor sounds a bit weird imho.